website/blog/2023-09-17-databend-weekly-111.md

```diff
@@ -0,0 +1,157 @@
+
+To enable this feature, you need to use version `v1.2.116-nightly` or later and enable UDF Server support in the configuration of `databend-query`:
+
+```toml
+[query]
+...
+enable_udf_server = true
+# use your udf server address here
+udf_server_allow_list = ['http://0.0.0.0:8815']
+...
+```
+
+The following code snippet demonstrates how to create a UDF Server in Python for calculating the greatest common divisor.
+
+```python
+from udf import *
+
+# Define a function that accepts nullable values, and set skip_null to True to enable it returns NULL if any argument is NULL.
+@udf(
+    input_types=["INT", "INT"],
+    result_type="INT",
+    skip_null=True,
+)
+def gcd(x: int, y: int) -> int:
+    while y != 0:
+        (x, y) = (y, x % y)
+    return x
+
+if __name__ == '__main__':
+    # create a UDF server listening at '0.0.0.0:8815'
+    server = UdfServer("0.0.0.0:8815")
+    # add defined functions
+    server.add_function(gcd)
+    # start the UDF server
+    server.serve()
+```
+
+In Databend, you can register UDF using the following SQL statement.
+
+```SQL
+CREATE FUNCTION gcd (INT, INT) RETURNS INT LANGUAGE python HANDLER = 'gcd' ADDRESS = 'http://0.0.0.0:8815';
+```
+
+This feature is currently in the preview stage. We offer a Python SDK file for demonstration and testing purposes. More SDKs will be released in the future. Feel free to join the UDF ecosystem.
+
+If you are interested in learning more, please check out the resources listed below:
+
+- [PR #12802 | feat: implement udf server in databend](https://github.com/datafuselabs/databend/issues/12802)
+- [Docs | UDF Server](https://github.com/datafuselabs/databend/blob/main/tests/udf-server/README.md)
+
+## Highlights
+
+We have also made these improvements to Databend that we hope you will find helpful:
+
+- Added support for infer filter.
+- Added support for idempotent deletion.
+- Supported the generation of aggregate functions, scalar functions, and expressions for SQLSmith testing.
+- Read the documentation [Docs | INSERT INTO](https://databend.rs/doc/sql-commands/dml/dml-insert) and [Docs | REPLACE INTO](https://databend.rs/doc/sql-commands/dml/dml-replace) to learn how to use SQL statements to insert data from Stage into a table.
+
+## What's Up Next
+
+We're always open to cutting-edge technologies and innovative ideas. You're more than welcome to join the community and bring them to Databend.
+
+### Providing Cache Support for Parquet Files in Stage
+
+Databend offers different types of caching to enhance query performance. These include Query Cache, File Metadata Cache, and Data Cache.
+
+However, there is currently a lack of effective cache support for files located in Stage. If metadata cache or object cache can be provided for Parquet files in Stage, it will help enhance the performance of querying external data.
```

Comment:
Currently, there exists an insufficient cache support mechanism for files residing in a stage. Introducing metadata caching or object caching for Parquet files within a stage could significantly boost query performance when working with external data.